library/Fission/Types.hs

```diff
@@ -563,3 +567,30 @@ instance Domain.Creator Fission where
   create domainName userId zoneId now =
     runDB $ Domain.create domainName userId zoneId now
 
+instance Challenge.Creator Fission where
+  create email = 
+    runDB $ Challenge.create email
+
+instance Challenge.Verifier Fission where
+  verify challenge = 
+    runDB $ Challenge.verify challenge
+
+instance MonadEmail Fission where
+  sendVerificationEmail recipient@(Email.Recipient { name }) challenge = do
```

Comment:
Protip: I’m 99.9999% sure that you don’t need the parens after the `@` because the record syntax keeps the fields “in” the constructor:

```haskell
recipient@Email.Recipient {name}
```

Reply:
yup you right :ok_hand: